packit/config/package_config.py

```diff
@@ -254,6 +255,65 @@ def __eq__(self, other: object):
         )
 
 
+class PackageConfigValidation:
+    def __init__(self, package_config: PackageConfig):
+        from packit.schema import PackageConfigSchema
+
+        self.schema_validation_errors: Dict = {}
+        self.validation_errors: Dict = {}
+
+        config_dict = vars(package_config)
+        del config_dict["_downstream_project_url"]
+        del config_dict["dist_git_clone_path"]
+        for field_name, errors in PackageConfigSchema().validate(config_dict).items():
+            if isinstance(errors, dict) and errors.get("_schema"):
+                self.schema_validation_errors[field_name] = errors["_schema"]
+            else:
+                self.validation_errors[field_name] = errors
+
+        if (
+            package_config.specfile_path
+            and not Path(package_config.specfile_path).is_file()
+        ):
+            self.validation_errors.setdefault("specfile_path", []).append(
+                f"{package_config.specfile_path} file does not exist"
+            )
+
+        for file_path in package_config.synced_files.files_to_sync:
+            if not Path(file_path.dest).is_file():
+                self.validation_errors.setdefault("synced_files", []).append(
+                    f"{file_path} file does not exist"
+                )
+
+        full_downstream_url = "https://src.fedoraproject.org/rpms/{0}/raw/master/f/{0}.spec".format(
+            package_config.downstream_package_name
+        )
+        if not requests.head(full_downstream_url):
+            self.validation_errors.setdefault("downstream_package_name", []).append(
+                f"Downstream package {package_config.downstream_package_name} does not exist"
+            )
+
+    def __repr__(self) -> str:
+        schema_validation_errors = ""
+        validation_errors = ""
+        for property_name, validation_output in self.schema_validation_errors.items():
+            if validation_output:
+                schema_validation_errors += (
+                    f"'{property_name} schema errors: {validation_output}', "
+                )
+        for property_name, validation_output in self.validation_errors.items():
+            if validation_output:
+                validation_errors += (
+                    f"'{property_name} validation errors: {validation_output}', "
+                )
+
+        return (
+            "PackageConfigValidation("
+            f"Schema Validation Errors({schema_validation_errors})"
+            f"Validation Errors({validation_errors}))"
+        )
```

Comment:
Any ideas what formatting should look like, this was created just to test and better understand PackageConfigSchema().validate.

Reply:
There are a few options here:
1. We can keep this in __repr__ of PackageConfig, but I feel like it's just too much for that method
2. Instead I'd like to see this in a dedicated method, something like `get_validation_errors_str`
3. Or, we could create a new class, which would contain the errors as attributes and we would process it in `cli/`